aws/src/test/java/org/apache/iceberg/aws/s3/signer/TestS3RestSigner.java

```diff
@@ -171,12 +171,8 @@ private static Server initHttpServer() throws Exception {
         new S3SignerServlet(S3ObjectMapper.mapper(), ImmutableList.of(deleteObjectsWithBody));
     ServletContextHandler servletContext =
         new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
-    servletContext.setContextPath("/");
-    ServletHolder servletHolder = new ServletHolder(servlet);
-    servletHolder.setInitParameter("javax.ws.rs.Application", "ServiceListPublic");
-    servletContext.addServlet(servletHolder, "/*");
-    servletContext.setVirtualHosts(null);
-    servletContext.setGzipHandler(new GzipHandler());
+    servletContext.addServlet(new ServletHolder(servlet), "/*");
+    servletContext.setHandler(new GzipHandler());
```

Comment:
`setGzipHandler` is deprecated, so switching to `setHandler` here and in other places

Reply:
Yeah I realized setting the Gzip handler is important, I thought it was already done by default but apparently not!